lib/src/async-compiler.ts

```diff
@@ -6,42 +6,144 @@ import {spawn} from 'child_process';
 import {Observable} from 'rxjs';
 import {takeUntil} from 'rxjs/operators';
 
+import {
+  OptionsWithLegacy,
+  StringOptionsWithLegacy,
+  createDispatcher,
+  handleCompileResponse,
+  handleLogEvent,
+  newCompilePathRequest,
+  newCompileStringRequest,
+} from './compiler';
 import {compilerCommand} from './compiler-path';
+import {FunctionRegistry} from './function-registry';
+import {ImporterRegistry} from './importer-registry';
+import * as utils from './utils';
+import * as proto from './vendor/embedded_sass_pb';
+import {CompileResult} from './vendor/sass';
 
 /**
- * An asynchronous wrapper for the embedded Sass compiler that exposes its stdio
- * streams as Observables.
+ * An asynchronous wrapper for the embedded Sass compiler
  */
-export class AsyncEmbeddedCompiler {
+export class AsyncCompiler {
   /** The underlying process that's being wrapped. */
   private readonly process = spawn(
     compilerCommand[0],
     [...compilerCommand.slice(1), '--embedded'],
     {windowsHide: true}
   );
 
+  /** Whether the underlying compiler has already exited. */
+  private disposed = false;
+
+  /** A record of all compilations */
+  private compilations: Promise<CompileResult>[] = [];
```

Comment:
This should be a set from which we remove the promises after they complete. Otherwise, this will leak memory during long-running applications.

Reply:
Added in ba7c156d2b0301974676e43f6680e74e635111e7